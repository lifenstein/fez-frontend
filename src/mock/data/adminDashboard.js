export const adminDashboardConfig = {
    admin_users: [
        {id: 123, name: 'Michael Feeney'},
        {id: 23, name: 'Another Staff'},
        {id: 444, name: 'Elizabeth Alvey'},
        {id: 2333, name: 'Lee Sibbald'},
    ],
};

export const adminDashboardToday = {
    systemalerts: {
        total: 150,
        today: 25,
        assigned: 15,
        unassigned: 135,
        mine: 8,
    },
    works: {
        processed: 82,
        unprocessed: 15,
    },
    oa: {
        current: 256,
        total: 1256,
    },
};

export const adminDashboardQuickLinks =  [
    { qlk_id: 1, qlk_amount: 10, qlk_order: 1, qlk_title: '2021+ Imported Records with an Author ID and Research Subtypes Only', qlk_link: 'https://espace.library.uq.edu.au/records/search?searchQueryParams%5ball%5d=%28author_id:%5b1+TO+%2a%5d%29+AND+%28date:%5b2021%5c-01%5c-01T00%5c:00%5c:00Z+TO+2024%5c-12%5c-31T00%5c:00%5c:00Z+%5d%29+AND+%28subtype:%28%22Article%22%29+OR+subtype:%28%22Critical%22%29+OR+subtype:%28%22Fully%22%29+OR+subtype:%28%22Research+Book%22%29+OR+subtype:%28%22Research+Book+Chapter%22%29%29+AND+%28%28isi_loc:%5b%2a+TO+%2a%5d%29+OR+%28scopus_id:%5b%2a+TO+%2a%5d%29%29+AND+%28%28ismemberof:+UQ:639325%29+OR+%28ismemberof:+UQ:308300%29+OR+%28ismemberof:+UQ:180159%29+OR+%28ismemberof:+UQ:688533%29+OR+%28ismemberof:+UQ:727866%29%29&page=1&pageSize=20&sortBy=score&sortDirection=Desc' },
    { qlk_id: 7, qlk_amount: null, qlk_order: 6, qlk_title: 'UQ Library', qlk_link: 'https://www.library.uq.edu.au' },
    { qlk_id: 2, qlk_amount: 30, qlk_order: 7, qlk_title: '2022+ Imported Records with no Author ID with subtype exclusions', qlk_link: 'https://espace.library.uq.edu.au/records/search?searchQueryParams%5ball%5d=%28%21author_id:%5b1+TO+%2a%5d%29+AND+%28date:%5b2022%5c-01%5c-01T00%5c:00%5c:00Z+TO+2024%5c-12%5c-31T00%5c:00%5c:00Z%5d%29+AND+%28%21subtype:%28%22letter%22%29+OR+%21subtype:%28%22published+abstract%22%29+OR+%21subtype:%28%22review+of+book%22%29+OR+%21subtype:%28%22introduction%22%29+OR+%21subtype:%28%22poster%22%29%29+%28%28ismemberof:+UQ:639325%29+OR+%28ismemberof:+UQ:308300%29+OR+%28ismemberof:+UQ:180159%29+OR+%28ismemberof:+UQ:688533%29+OR+%28ismemberof:+UQ:727866%29%29&page=1&pageSize=20&sortBy=score&sortDirection=Desc' },
    { qlk_id: 3, qlk_amount: 60, qlk_order: 2, qlk_title: '2020 - 2023 Imported Records with an Author ID and Research Subtypes Only', qlk_link: 'https://espace.library.uq.edu.au/records/search?searchQueryParams%5ball%5d=%28author_id:%5b1+TO+%2a%5d%29+AND+%28date:%5b2020%5c-01%5c-01T00%5c:00%5c:00Z+TO+2023%5c-12%5c-31T00%5c:00%5c:00Z+%5d%29+AND+%28subtype:%28%22Article%22%29+OR+subtype:%28%22Critical%22%29+OR+subtype:%28%22Fully%22%29+OR+subtype:%28%22Research+Book%22%29+OR+subtype:%28%22Research+Book+Chapter%22%29%29+AND+%28%28isi_loc:%5b%2a+TO+%2a%5d%29+OR+%28scopus_id:%5b%2a+TO+%2a%5d%29%29+AND+%28%28ismemberof:+UQ:639325%29+OR+%28ismemberof:+UQ:308300%29+OR+%28ismemberof:+UQ:180159%29+OR+%28ismemberof:+UQ:688533%29+OR+%28ismemberof:+UQ:727866%29%29&page=1&pageSize=20&sortBy=score&sortDirection=Desc' },
    { qlk_id: 4, qlk_amount: 90, qlk_order: 3, qlk_title: '2021 - 2023 Imported Records with no Author ID with subtype exclusions', qlk_link: 'https://espace.library.uq.edu.au/records/search?searchQueryParams%5ball%5d=%28%21author_id:%5b1+TO+%2a%5d%29+AND+%28date:%5b2021%5c-01%5c-01T00%5c:00%5c:00Z+TO+2023%5c-12%5c-31T00%5c:00%5c:00Z%5d%29+AND+%28%21subtype:%28%22letter%22%29+OR+%21subtype:%28%22published+abstract%22%29+OR+%21subtype:%28%22review+of+book%22%29+OR+%21subtype:%28%22introduction%22%29+OR+%21subtype:%28%22poster%22%29%29+%28%28ismemberof:+UQ:639325%29+OR+%28ismemberof:+UQ:308300%29+OR+%28ismemberof:+UQ:180159%29+OR+%28ismemberof:+UQ:688533%29+OR+%28ismemberof:+UQ:727866%29%29+AND+%28%21institutional_status:%22Non-UQ%22%29&page=1&pageSize=20&sortBy=score&sortDirection=Desc' },
    { qlk_id: 5, qlk_amount: 110, qlk_order: 4, qlk_title: 'Not yet publicly available (with merged metadata)', qlk_link: 'https://espace.library.uq.edu.au/records/search?searchQueryParams%5ball%5d=%28isMemberOf:+UQ:676357%29+AND+%28%28doi:%5b%2a+TO+%2a%5d%29+OR+%28isi_loc:%5b%2a+TO+%2a%5d%29+OR+%28scopus_id:%5b%2a+TO+%2a%5d%29%29&page=1&pageSize=20&sortBy=score&sortDirection=Desc' },
    { qlk_id: 6, qlk_amount: 2345, qlk_order: 5, qlk_title: 'eSpace Followup', qlk_link: 'https://espace.library.uq.edu.au/records/search?page=1&pageSize=20&sortBy=score&sortDirection=Desc&searchQueryParams%5ball%5d%5bvalue%5d=ismemberof:+UQ:237927&searchQueryParams%5ball%5d%5blabel%5d=&searchMode=advanced' },
];

export const adminDashboardSystemAlerts = [
    {sat_id:1, sat_title: 'My Works - Claimed Work - UQ:8efd126 - uqwtomas', sat_content: 'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.', sat_link: 'https://espace.library.uq.edu.au/view/UQ:efd126', sat_created_date: '3/4/2024 15:55:00', sat_assigned_date: null, sat_resolved_date: undefined, sat_assigned_to: undefined, sat_resolved_by: undefined },
    {sat_id:12, sat_title: 'Issues on record - UQ:34555c6', sat_content: 'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.', sat_link: 'https://espace.library.uq.edu.au/view/UQ:efd126', sat_created_date: '3/4/2024 15:55:00', sat_assigned_date: null, sat_resolved_date: undefined, sat_assigned_to: 123, sat_resolved_by: undefined },
    {sat_id:13, sat_title: 'My Works - Claimed Work - UQ:1494946 - uqmdeben', sat_content: 'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.', sat_link: 'https://espace.library.uq.edu.au/view/UQ:efd126', sat_created_date: '3/4/2024 15:55:00', sat_assigned_date: null, sat_resolved_date: undefined, sat_assigned_to: 23, sat_resolved_by: undefined },
    {sat_id:14, sat_title: '(UQ eSpace) Keywords with full stops at the end', sat_content: 'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.', sat_link: 'https://espace.library.uq.edu.au/view/UQ:efd126', sat_created_date: '3/4/2024 15:55:00', sat_assigned_date: null, sat_resolved_date: undefined, sat_assigned_to: undefined, sat_resolved_by: undefined },
    {sat_id:51, sat_title: 'My Works - Disowned Work - UQ:4128e45 - uqsmcor3', sat_content: 'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.', sat_link: 'https://espace.library.uq.edu.au/view/UQ:efd126', sat_created_date: '3/4/2024 15:55:00', sat_assigned_date: null, sat_resolved_date: undefined, sat_assigned_to: 444, sat_resolved_by: undefined },
    {sat_id:16, sat_title: 'My Works - Claimed Work - UQ:e9a2910 - uqnpacha', sat_content: 'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.', sat_link: 'https://espace.library.uq.edu.au/view/UQ:efd126', sat_created_date: '3/4/2024 15:55:00', sat_assigned_date: null, sat_resolved_date: undefined, sat_assigned_to: undefined, sat_resolved_by: undefined },
    {sat_id:71, sat_title: 'My Works - Disowned Work - UQ:02daQbe - uqrelli6', sat_content: 'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.', sat_link: 'https://espace.library.uq.edu.au/view/UQ:efd126', sat_created_date: '3/4/2024 15:55:00', sat_assigned_date: null, sat_resolved_date: undefined, sat_assigned_to: undefined, sat_resolved_by: undefined },
    {sat_id:188, sat_title: 'My Works - Claimed Work - UQ:4489ca8 - uqwtomas', sat_content: 'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.', sat_link: 'https://espace.library.uq.edu.au/view/UQ:efd126', sat_created_date: '3/4/2024 15:55:00', sat_assigned_date: null, sat_resolved_date: undefined, sat_assigned_to: 2333, sat_resolved_by: undefined },
];

export const adminDashboardReportWorksData = [
    {
        id: 1,
        pid: 'UQ:abc123',
        date_created: '9/4/2024 10:44',
        work_type: 'Journal Article - Article (original research)',
        user: 'uqmfeene',
        topic: 'Updated record by O’Keefe, Christopher J, (User ID: 64567)',
    },
    {
        id: 2,
        pid: 'UQ:abc123',
        date_created: '9/4/2024 10:44',
        work_type: 'Journal Article - Article (original research)',
        user: 'uqmfeene',
        topic: 'Updated record by O’Keefe, Christopher J, (User ID: 64567)',
    },
    {
        id: 3,
        pid: 'UQ:abc123',
        date_created: '9/4/2024 10:44',
        work_type: 'Journal Article - Article (original research)',
        user: 'uqmfeene',
        topic: 'Updated record by O’Keefe, Christopher J, (User ID: 64567)',
    },
    {
        id: 4,
        pid: 'UQ:abc123',
        date_created: '9/4/2024 10:44',
        work_type: 'Journal Article - Article (original research)',
        user: 'uqmfeene',
        topic: 'Updated record by O’Keefe, Christopher J, (User ID: 64567)',
    },
    {
        id: 5,
        pid: 'UQ:abc123',
        date_created: '9/4/2024 10:44',
        work_type: 'Journal Article - Article (original research)',
        user: 'uqmfeene',
        topic: 'Updated record by O’Keefe, Christopher J, (User ID: 64567)',
    },
    {
        id: 6,
        pid: 'UQ:abc123',
        date_created: '9/4/2024 10:44',
        work_type: 'Journal Article - Article (original research)',
        user: 'uqmfeene',
        topic: 'Updated record by O’Keefe, Christopher J, (User ID: 64567)',
    },
];

export const adminDashboardReportSystemAlertsData = [
{
    id: 1,
    date_created: '9/4/2024 10:44',
    assigned_to: 'Michael Feeney',
    assigned_date: '9/4/2024 10:44',
    resolved_by: 'Elizabeth Alvey',
    resolved_date: '9/4/2024 10:44',
    title: 'My Works - Claimed Work - UQ:8efd126 - uqwtomas',
    content:
        'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.',
    link: 'https://espace.library.uq.edu.au/view/UQ:efd126',
},
{
    id: 2,
    date_created: '9/4/2024 10:44',
    assigned_to: 'Michael Feeney',
    assigned_date: '9/4/2024 10:44',
    resolved_by: 'Elizabeth Alvey',
    resolved_date: '9/4/2024 10:44',
    title: 'My Works - Claimed Work - UQ:8efd126 - uqwtomas',
    content:
        'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.',
    link: 'https://espace.library.uq.edu.au/view/UQ:efd126',
},
{
    id: 3,
    date_created: '9/4/2024 10:44',
    assigned_to: 'Michael Feeney',
    assigned_date: '9/4/2024 10:44',
    resolved_by: 'Elizabeth Alvey',
    resolved_date: '9/4/2024 10:44',
    title: 'My Works - Claimed Work - UQ:8efd126 - uqwtomas',
    content:
        'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.',
    link: 'https://espace.library.uq.edu.au/view/UQ:efd126',
},
{
    id: 4,
    date_created: '9/4/2024 10:44',
    assigned_to: 'Michael Feeney',
    assigned_date: '9/4/2024 10:44',
    resolved_by: 'Elizabeth Alvey',
    resolved_date: '9/4/2024 10:44',
    title: 'My Works - Claimed Work - UQ:8efd126 - uqwtomas',
    content:
        'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.',
    link: 'https://espace.library.uq.edu.au/view/UQ:efd126',
},
{
    id: 5,
    date_created: '9/4/2024 10:44',
    assigned_to: 'Michael Feeney',
    assigned_date: '9/4/2024 10:44',
    resolved_by: 'Elizabeth Alvey',
    resolved_date: '9/4/2024 10:44',
    title: 'My Works - Claimed Work - UQ:8efd126 - uqwtomas',
    content:
        'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.',
    link: 'https://espace.library.uq.edu.au/view/UQ:efd126',
},
{
    id: 6,
    date_created: '9/4/2024 10:44',
    assigned_to: 'Michael Feeney',
    assigned_date: '9/4/2024 10:44',
    resolved_by: 'Elizabeth Alvey',
    resolved_date: '9/4/2024 10:44',
    title: 'My Works - Claimed Work - UQ:8efd126 - uqwtomas',
    content:
        'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.',
    link: 'https://espace.library.uq.edu.au/view/UQ:efd126',
},
{
    id: 7,
    date_created: '9/4/2024 10:44',
    assigned_to: 'Michael Feeney',
    assigned_date: '9/4/2024 10:44',
    resolved_by: 'Elizabeth Alvey',
    resolved_date: '9/4/2024 10:44',
    title: 'My Works - Claimed Work - UQ:8efd126 - uqwtomas',
    content:
        'User “Tomaszewski, Wojciech (Wojtek) (uqwtomas)” has indicated that they are the author “Tomaszewski, Wojciech (Wojtek) (87054)” on this work.',
    link: 'https://espace.library.uq.edu.au/view/UQ:efd126',
}];