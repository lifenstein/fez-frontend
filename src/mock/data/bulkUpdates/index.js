export { default as bulkUpdatesList } from './bulkUpdatesList';