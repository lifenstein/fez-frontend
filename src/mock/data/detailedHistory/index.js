export { default as detailedHistory } from './detailedHistoryList';