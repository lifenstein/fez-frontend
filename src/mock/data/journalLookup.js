export const journalLookup = {
    total: "3",
    data: [
        {
            "jnl_jid": 1,
            "jnl_title": "Abstract and Applied Analysis",
            "jnl_publisher": "Publisher",
            "jnl_created_date": "2010-11-19",
            "jnl_updated_date": "2010-11-19",
            "fez_journal_issn": [
                {
                    "jnl_issn_jid": 1,
                    "jnl_issn": "1085-3375",
                    "jnl_issn_order": 1,
                },
                {
                    "jnl_issn_jid": 1,
                    "jnl_issn": "1687-0409",
                    "jnl_issn_order": 2,
                },
            ],
        },
        {
            "jnl_jid": 2,
            "jnl_title": "Acta Mathematica Vietnamica",
            "jnl_publisher": "Publisher 2",
            "jnl_created_date": "2011-01-19",
            "jnl_updated_date": "2011-01-19",
            "fez_journal_issn": [
                {
                    "jnl_issn_jid": 2,
                    "jnl_issn": "0251-4184",
                    "jnl_issn_order": 1,
                },
            ],
        },
        {
            "jnl_jid": 3,
            "jnl_title": "Journal of Physics D: Applied Physics",
            "jnl_publisher": "Publisher 1",
            "jnl_created_date": "2011-01-19",
            "jnl_updated_date": "2011-01-19",
            "fez_journal_issn": [
                {
                    "jnl_issn_jid": 3,
                    "jnl_issn": "0251-4184",
                    "jnl_issn_order": 1,
                },
                {
                    "jnl_issn_jid": 3,
                    "jnl_issn": "1361-6463",
                    "jnl_issn_order": 1,
                },
            ],
        },
    ],
};