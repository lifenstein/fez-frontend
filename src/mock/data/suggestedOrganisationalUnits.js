export const suggestedOrganisationalUnits = {
    "data": [
        {
            "aut_id": 7624839,
            "org_title": "Information Systems and Resource Services (University of Queensland Library)",
            "org_id": 1248
        }
    ]
};