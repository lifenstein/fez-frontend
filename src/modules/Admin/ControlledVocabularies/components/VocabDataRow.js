import React from 'react';
import PropTypes from 'prop-types';

import IconButton from '@mui/material/IconButton';
import Box from '@mui/material/Box';
import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
import Grid from '@mui/material/Grid';

export const VocabDataRow = ({ row }) => {
    return (
        <Grid
            container
            key={row.cvo_id}
            data-testid={`row-${row.cvo_id}`}
            sx={{ boxSizing: 'border-box', boxShadow: '0 -1px 0 #eaeaea', padding: '15px 0px 0px' }}
        >
            <React.Fragment key={row.cvo_id}>
                <Grid container sx={{ paddingBottom: '10px' }}>
                    <Grid item xs={1} sm={1} md={1}>
                        <Box sx={{ float: 'left', width: '24px' }}>
                            <IconButton
                                sx={{ paddingTop: '5px' }}
                                aria-label="expand row"
                                size="small"
                                id={`expand-row-${row.cvo_id}`}
                                data-analyticsid={`expand-row-${row.cvo_id}`}
                                data-testid={`expand-row-${row.cvo_id}`}
                            >
                                {<KeyboardArrowDownIcon />}
                            </IconButton>
                        </Box>
                    </Grid>
                    <Grid item md={8} sm={6} xs={6}>
                        <Box>{row.cvo_title}</Box>
                    </Grid>
                    <Grid item md={1} xs={2} sm={2}>
                        <Box>{/* row.cvo_image_filename*/}</Box>
                    </Grid>
                    <Grid item md={1} xs={2} sm={2}>
                        <Box>{row.cvo_external_id}</Box>
                    </Grid>
                    <Grid item md={1} xs={1} sm={1}>
                        {''}
                    </Grid>
                </Grid>
            </React.Fragment>
        </Grid>
    );
};
VocabDataRow.propTypes = {
    conf: PropTypes.object,
    row: PropTypes.object,
    adminUser: PropTypes.bool,
    labels: PropTypes.object,
};
export default VocabDataRow;
