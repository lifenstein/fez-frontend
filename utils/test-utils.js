/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { Route } from 'react-router';
import { mui1theme } from 'config/theme';
import { Provider } from 'react-redux';
import { ThemeProvider as MuiThemeProvider } from '@material-ui/core/styles';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import MomentUtils from '@date-io/moment';

import { getStore } from '../src/config/store';
import Immutable from 'immutable';
import { createMemoryHistory } from 'history';

const domTestingLib = require('@testing-library/dom');
const reactTestingLib = require('@testing-library/react');
const mime = require('mime-types');

const { configure } = domTestingLib;

configure(config => ({
    ...config,
    testIdAttribute: 'id',
}));

export const AllTheProviders = props => {
    return (
        <MuiThemeProvider theme={mui1theme}>
            <MuiPickersUtilsProvider utils={MomentUtils}>{props.children}</MuiPickersUtilsProvider>
        </MuiThemeProvider>
    );
};

AllTheProviders.propTypes = {
    children: PropTypes.node,
};

export const rtlRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export const renderWithRouter = (
    ui,
    { route = '/', history = createMemoryHistory({ initialEntries: [route] }), renderMethod = render } = {},
) => {
    return renderMethod(
        <AllTheProviders>
            <Router history={history}>{ui}</Router>
        </AllTheProviders>,
    );
};

export const WithRouter = ({ children, route = '/', history = createMemoryHistory({ initialEntries: [route] }) }) => (
    <AllTheProviders>
        <Router history={history}>{children}</Router>
    </AllTheProviders>
);
export const renderWithRedux = ({ initialState }) => render => {
    return {
        ...render,
        store: getStore({ initialState, history: render.history }),
    };
};

export const withRouter = ({
    route = '/',
    path = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
} = {}) => WrappedComponent => {
    return (
        <Router history={history}>
            <Route path={path} children={WrappedComponent} />
        </Router>
    );
};

export const withRedux = (initialState = Immutable.Map()) => WrappedComponent => {
    return <Provider store={getStore(initialState)}>{WrappedComponent}</Provider>;
};

export const WithReduxStore = ({ initialState = Immutable.Map(), children }) => (
    <Provider store={getStore(initialState)}>
        <AllTheProviders>{children}</AllTheProviders>
    </Provider>
);

export const defaultDatastream = {
    dsi_embargo_date: null,
    dsi_open_access: 1,
    dsi_label: '',
    dsi_copyright: null,
    dsi_state: 'A',
    dsi_size: 27932352,
};
export const createFezDatastreamInfoArray = (datastreams, pid = null, withPreview = true) => {
    let processed = datastreams.filter(datastream => typeof datastream === 'object' || typeof datastream === 'string');
    if (withPreview) {
        processed = [];
        datastreams.forEach(function(datastream) {
            processed.push(datastream);
            const filename = typeof datastream === 'object' ? datastream.dsi_dsid : datastream;
            const mimetype = mime.lookup(filename);
            // bail in case it's not derivable
            if (
                !filename.includes('.pdf') &&
                !mimetype.includes('image') &&
                !mimetype.includes('audio') &&
                !mimetype.includes('video')
            ) {
                return;
            }
            // eslint-disable-next-line no-nested-ternary
            const derivativeMimetype = mimetype.includes('audio')
                ? 'audio/mp3'
                : mimetype.includes('video')
                ? 'video/mp4'
                : 'image/jpg';

            (derivativeMimetype.includes('image') ? ['preview_', 'thumbnail_', 'web_'] : ['']).forEach(function(
                prefix,
            ) {
                processed.push({
                    dsi_dsid: `${prefix}${filename.split('.')[0]}_t.${derivativeMimetype.split('/').pop()}`,
                    dsi_mimetype: derivativeMimetype,
                });
            });
        });
    }

    return processed.map((datastream, index) => ({
        ...defaultDatastream,
        ...{
            dsi_pid: pid,
            dsi_order: index + 1,
            dsi_mimetype: mime.lookup(typeof datastream === 'object' ? datastream.dsi_dsid : datastream),
        },
        ...(typeof datastream === 'object'
            ? datastream
            : {
                  dsi_dsid: datastream,
              }),
    }));
};

export const withDatastreams = (sources, datastreams, callback) =>
    datastreams.map(item => {
        const source = sources.find(datastream =>
            datastream.dsi_dsid.includes(
                item.dsi_dsid
                    .replace('_t.', '.')
                    .replace('preview_', '')
                    .replace('thumbnail_', '')
                    .replace('web_', '')
                    .split('.')[0],
            ),
        );
        const isDerivative = item.dsi_dsid !== source.dsi_dsid;
        callback(item, source, isDerivative);
    });

export const getDatastreamByFilename = (filename, derivatives) =>
    derivatives.find(derivative => derivative.dsi_dsid === filename);

module.exports = {
    ...domTestingLib,
    ...reactTestingLib,
    rtlRender,
    renderWithRouter,
    renderWithRedux,
    withRedux,
    withRouter,
    AllTheProviders,
    WithReduxStore,
    WithRouter,
    createFezDatastreamInfoArray,
    getDatastreamByFilename,
    withDatastreams,
};
