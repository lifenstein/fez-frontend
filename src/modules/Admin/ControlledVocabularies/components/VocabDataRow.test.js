import React from 'react';

import { render } from 'test-utils';

import * as mockData from 'mock/data';

import VocabDataRow from './VocabDataRow';

const vocabDataRow = mockData.vocabList.data[0];

const setup = vocabDataRow => render(<VocabDataRow row={vocabDataRow} />);

describe('ControlledVocabularies VocabDataRow', () => {
    it('should render the table', async () => {
        const myTable = setup(vocabDataRow);
        expect(myTable).toMatchSnapshot();
    });
});
