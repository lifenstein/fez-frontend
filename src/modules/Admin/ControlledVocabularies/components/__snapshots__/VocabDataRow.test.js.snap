// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlledVocabularies VocabDataRow should render the table 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container css-ccbfi6-MuiGrid-root"
        data-testid="row-453669"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-oonwph-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1 MuiGrid-grid-md-1 css-u78q51-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-rve04c"
            >
              <button
                aria-label="expand row"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1aw2z18-MuiButtonBase-root-MuiIconButton-root"
                data-analyticsid="expand-row-453669"
                data-testid="expand-row-453669"
                id="expand-row-453669"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyboardArrowDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-8 css-5oeiha-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              AIATSIS codes
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-2 MuiGrid-grid-md-1 css-1ifj80e-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-2 MuiGrid-grid-md-1 css-1ifj80e-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1 MuiGrid-grid-md-1 css-u78q51-MuiGrid-root"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container css-ccbfi6-MuiGrid-root"
      data-testid="row-453669"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-oonwph-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1 MuiGrid-grid-md-1 css-u78q51-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-rve04c"
          >
            <button
              aria-label="expand row"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1aw2z18-MuiButtonBase-root-MuiIconButton-root"
              data-analyticsid="expand-row-453669"
              data-testid="expand-row-453669"
              id="expand-row-453669"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-8 css-5oeiha-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            AIATSIS codes
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-2 MuiGrid-grid-md-1 css-1ifj80e-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-2 MuiGrid-grid-md-1 css-1ifj80e-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1 MuiGrid-grid-md-1 css-u78q51-MuiGrid-root"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
